components/apimgt/org.wso2.carbon.apimgt.common.gateway/src/main/java/org/wso2/carbon/apimgt/common/gateway/graphql/QueryAnalyzer.java

```diff
@@ -62,8 +80,18 @@ public QueryAnalyzerResponseDTO analyseQueryDepth(int maxQueryDepth, String payl
             MaxQueryDepthInstrumentation maxQueryDepthInstrumentation =
                     new MaxQueryDepthInstrumentation(maxQueryDepth);
             GraphQL runtime = GraphQL.newGraphQL(schema).instrumentation(maxQueryDepthInstrumentation).build();
+            if (variableMap == null) {
+                variableMap = new HashMap<>();
+            }
 
-            ExecutionResult executionResult = runtime.execute(payload);
+            ExecutionInput executionInput = ExecutionInput.newExecutionInput()
+                    .query(payload)
+                    .context(Optional.ofNullable(null))
+                    .root(null)
```

Comment:
we can remove `root(null)`

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12441/commits/fcd4966a4edf7516fe689a328717370308cc02e5